From 2661611d9fec803c0232911fa3ca421f336a933c Mon Sep 17 00:00:00 2001
From: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Date: Thu, 8 Jun 2023 11:34:02 +0200
Subject: [PATCH] arm64: dts: st: add cryp1 on stm32mp25

Add cryp1 node for stm32mp25xc and stm32mp25xf with DMA support.

Signed-off-by: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Change-Id: Ic66fcf22f24764ea81c44d4fca04d70658a55c1e
---
 arch/arm64/boot/dts/st/stm32mp25xc.dtsi | 14 +++++++++++++-
 arch/arm64/boot/dts/st/stm32mp25xf.dtsi | 14 +++++++++++++-
 2 files changed, 26 insertions(+), 2 deletions(-)

--- a/arch/arm64/boot/dts/st/stm32mp25xc.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp25xc.dtsi
@@ -4,5 +4,17 @@
  * Author: Alexandre Torgue <alexandre.torgue@foss.st.com> for STMicroelectronics.
  */
 
-/ {
+&rifsc {
+	cryp1: crypto@42030000 {
+		compatible = "st,stm32mp1-cryp";
+		reg = <0x42030000 0x1000>;
+		interrupts = <GIC_SPI 144 IRQ_TYPE_LEVEL_HIGH>;
+		clocks = <&rcc CK_BUS_CRYP1>;
+		resets = <&rcc CRYP1_R>;
+		dmas = <&hpdma 4 0x40 0x3021>,
+		       <&hpdma 5 0x43 0x3012>;
+		dma-names = "in", "out";
+		access-controllers = <&rifsc 96>;
+		status = "disabled";
+	};
 };
--- a/arch/arm64/boot/dts/st/stm32mp25xf.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp25xf.dtsi
@@ -4,5 +4,17 @@
  * Author: Alexandre Torgue <alexandre.torgue@foss.st.com> for STMicroelectronics.
  */
 
-/ {
+&rifsc {
+	cryp1: crypto@42030000 {
+		compatible = "st,stm32mp1-cryp";
+		reg = <0x42030000 0x1000>;
+		interrupts = <GIC_SPI 144 IRQ_TYPE_LEVEL_HIGH>;
+		clocks = <&rcc CK_BUS_CRYP1>;
+		resets = <&rcc CRYP1_R>;
+		dmas = <&hpdma 4 0x40 0x3021>,
+		       <&hpdma 5 0x43 0x3012>;
+		dma-names = "in", "out";
+		access-controllers = <&rifsc 96>;
+		status = "disabled";
+	};
 };
